src/atomate2/abinit/jobs/base.py

```diff
@@ -0,0 +1,815 @@
+
+        It will first try to link the fortran file and then the Netcdf file, if the first is not found.
+
+        Parameters
+        ----------
+        ext : str
+            extension that should be linked.
+        source_dir : str
+            path to the source directory.
+        strict : bool
+            whether to raise an exception if the file is missing.
+
+        Returns
+        -------
+        str
+            The path to the generated link. None if strict=False and the file could not be found.
+        """
+        source = os.path.join(source_dir, OUTDATA_PREFIX + "_" + ext)
+        logger.info("Need path {} with ext {}".format(source, ext))
+        dest = os.path.join(self.workdir, INDATA_PREFIX + "_" + ext)
+
+        if not os.path.exists(source):
+            # Try netcdf file. TODO: this case should be treated in a cleaner way.
+            source += ".nc"
+            if os.path.exists(source):
+                dest += ".nc"
+
+        if not os.path.exists(source):
+            if strict:
+                msg = "{} is needed by this job but it does not exist".format(source)
+                logger.error(msg)
+                raise InitializationError(msg)
+            return
+
+        # Link path to dest if dest link does not exist.
+        # else check that it points to the expected file.
+        logger.info("Linking path {} --> {}".format(source, dest))
+        if not os.path.exists(dest) or not strict:
+            if self.settings.COPY_DEPS:
+                shutil.copyfile(source, dest)
+            else:
+                os.symlink(source, dest)
+            return dest
+        else:
+            # check links but only if we haven't performed the restart.
+            # in this case, indeed we may have replaced the file pointer with the
+            # previous output file of the present job.
+            if (
+                not self.settings.COPY_DEPS
+                and os.path.realpath(dest) != source
+                # and not self.restart_info
+            ):
+                msg = "dest {} does not point to path {}".format(dest, source)
+                logger.error(msg)
+                raise InitializationError(msg)
+
+    def resolve_restart_deps(self, prev_dir):
+        """Resolve dependencies for a job that is restarted."""
+        # To be implemented for specific types of job (a relaxation is not restarted the same way
+        # an scf is, or a non scf, or ...)
+
+    @staticmethod
+    def set_logger():
+        """Set a logger for pymatgen.io.abinit and abipy."""
+        # TODO: what to do here ?
+        # log_handler = logging.FileHandler('abipy.log')
+        # log_handler.setFormatter(logging.Formatter(logging.BASIC_FORMAT))
+        # logging.getLogger('pymatgen.io.abinit').addHandler(log_handler)
+        # logging.getLogger('abipy').addHandler(log_handler)
+        # logging.getLogger('abiflows').addHandler(log_handler)
+
+    def set_workdir(self, workdir):
```

Comment:
Should this be handled by the input set generator?

Reply:
yes, done